### DISCUSSION TITLE
php PDO connection

### DISCUSSION THREAD

**Discussion by hubertz** [planetscale/discussion#376] [Q&A] [answered]
im trying to connect with vanilla php(xampp/windows ) to planetscale  I been trying in so many different ways but i cannot make it connect.

first: the documentation for windows certificate is broken: ->
https://github.com/planetscale/connection-examples/blob/main/concepts/secure-connections

second: i got this error:
No connection could be made because the target machine actively refused it in

i tried the .env method ,different types of cacert.pems, and different types of certificates

**Comment by fooforge** [comment, marked as answer]
Hi @hubertz,

Thanks for letting us know about the broken link to the docs. This has been fixed in planetscale/connections-example#14; the correct link is [this one](https://planetscale.com/docs/concepts/secure-connections#windows).

In a nut-shell, what you will need to do is to download the latest CA bundle from https://curl.se/docs/caextract.html, store it somewhere where your application can access the file, and point your database driver to it. That should be it.

If you still can't get it to run, I would need to see some code examples, e.g. your database driver's configuration and your connection string. Please make sure to obscure any secrets such as your database name, username and password.

**Comment by hubertz** [comment]
`this` is my:  connection `string`

`<?php`
$hostname = $_ENV['HOST'];
$dbName = $_ENV['DATABASE'];
$username = $_ENV['USERNAME'];
$password = $_ENV['PASSWORD'];
$ssl = `$_ENV['MYSQL_ATTR_SSL_CA'];`

$mysqli = mysqli_init();
$mysqli->ssl_set(NULL, NULL, $ssl, NULL, NULL);
$mysqli->real_connect($hostname, $username, $password, $dbName, $port);

if ($mysqli->connect_error) {
    echo 'not connected to the database';
} else {
    echo "Connected successfully";
}


and this is  my env.
HOST=secret
USERNAME=secret
PASSWORD=secret
DATABASE=secretdatabase
MYSQL_ATTR_SSL_CA=C:\xampp2\htdocs\cacert.pem

**Reply by hubertz** [reply]
Also i got this error "Warning: mysqli::real_connect(): (HY000/1045): Access denied for user ''@'localhost' (using password: NO) in C:\xampp3\htdocs\wop\config\connection2.php on line 10
not connected to the database"
